File: 2-ad-hoc/2387.py
n = int(input())
consultas = [{'inicio': 0, 'fim': 0} for i in range(n)]

for i in range(n):
    inicio, fim = map(int, input().split())
    consultas[i]['inicio'] = inicio
    consultas[i]['fim'] = fim

consultas.sort(key=lambda x: x['fim'])

qtd = 1
fim = consultas[0]['fim']

for i in range(1, n):
    if consultas[i]['inicio'] >= fim:
        qtd += 1
        fim = consultas[i]['fim']

print(qtd)

File: C++/2-ad-hoc/2387.cpp
#include <algorithm>
#include <iostream>
#define MAX 10000

using namespace std;

struct Consulta {
    int inicio;
    int fim;
} consultas[MAX];

bool compara(Consulta a, Consulta b) {
    return a.fim < b.fim;
}

int main() {
    int n, qtd = 1;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> consultas[i].inicio >> consultas[i].fim;
    }

    sort(consultas, consultas + n, compara);

    int fim = consultas[0].fim;

    for (int i = 1; i < n; i++) {
        if (consultas[i].inicio >= fim) {
            fim = consultas[i].fim;
            qtd++;
        }
    }

    cout << qtd << endl;
}